tools/build-tasks/src/rush.ts

```diff
@@ -26,7 +26,7 @@ export class RushContext {
   private readonly packageStore: string;
 
   constructor(public readonly rushConfig: RushConfiguration) {
-    this.packageStore = path.join(rushConfig.pnpmStoreFolder, '2');
+    this.packageStore = path.join(rushConfig.pnpmOptions.pnpmStorePath, '2');
```

Comment:
@dbartol the problem is here, I believe. `pnpmStorePath` is not correct. I haven't looked too deeply here, but I just think the `packageStore` path is not correct. Is `2` a version number>

Reply:
The existing deployment code depends on the layout of the pnpm store and shrinkwrap file. The root of the problem is that there seems to be no good way to get from the maze of symlinks created by pnpm to the npm-style nested `node_modules` layout expected by the `vsce` tool that packages the extension. We currently try to find the right packages in the pnpm package store and copy them into a `node_modules` tree, allowing duplicates that would normally by symlinked by pnpm. Then we run `npm dedupe` to try to hoist out the duplicates before we run `vsce package`.

There seems to be a new `rush deploy` command in the most recent releases of Rush. This does part of what we need, by copying the package and its dependencies to a deployment directory, skipping any dev dependencies. However, even this deployment directory uses pnpm's symlinking strategy, because to do otherwise would potentially break the semantics in subtle ways.

I wonder if we'd be better off just switching package managers. Rush supports pnpm, npm, and Yarn. When I first compared them, pnpm sounded like the best solution, but if we lose one of its primary benefits when packaging our extension anyway, maybe it's not worth the trouble.

`vsce package` does have a `--yarn` switch now. I wonder if we could just switch to Yarn and get some of this for free.

Reply:
`pnpm` is the package manager best supported by rush. But, yeah...I see the benefits of pnpm aren't relevant for packaging the extension. I'll work on this some more and try swapping out for yarn.  Is it too late to move to lerna or yarn workspaces?